staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go

```diff
@@ -282,66 +296,59 @@ func (o *ExposeServiceOptions) RunExpose(cmd *cobra.Command, args []string) erro
 					return cmdutil.UsageErrorf(cmd, "couldn't find port via --port flag or introspection")
 				}
 			case 1:
-				params["port"] = ports[0]
+				o.Port = ports[0]
 			default:
-				params["ports"] = strings.Join(ports, ",")
+				o.Ports = strings.Join(ports, ",")
 			}
 		}
 
 		// Always try to derive protocols from the exposed object, may use
 		// different protocols for different ports.
-		if _, found := params["protocol"]; found {
-			protocolsMap, err := o.ProtocolsForObject(info.Object)
-			if err != nil {
-				return cmdutil.UsageErrorf(cmd, "couldn't find protocol via introspection: %v", err)
-			}
-			if protocols := generate.MakeProtocols(protocolsMap); !generate.IsZero(protocols) {
-				params["protocols"] = protocols
-			}
+		protocolsMap, err := o.ProtocolsForObject(info.Object)
+		if err != nil {
+			return cmdutil.UsageErrorf(cmd, "couldn't find protocol via introspection: %v", err)
+		}
+		if protocols := generate.MakeProtocols(protocolsMap); !generate.IsZero(protocols) {
+			o.Protocols = protocols
 		}
 
-		if generate.IsZero(params["labels"]) {
+		if len(o.Labels) == 0 {
 			labels, err := meta.NewAccessor().Labels(info.Object)
 			if err != nil {
 				return err
 			}
-			params["labels"] = polymorphichelpers.MakeLabels(labels)
-		}
-		if err = generate.ValidateParams(names, params); err != nil {
```

Comment:
`service/v2` has two required values: `default-name` and `selector` see https://github.com/kubernetes/kubectl/blob/1bf219da9bdd8c03758195f5b43f1ddfbe31c3e7/pkg/generate/versioned/service.go#L55-L57 make sure to have appropriate `Validate` method to reflect that.

Reply:
both of that was done in `createService()` , but will move it to `Validate()`

Reply:
`default-name` cannot be moved to `Validate` because if user does not provide a name it will call the API server and get from the `info` object.